src/otx/core/utils/build.py

```diff
@@ -40,18 +40,43 @@ def build_mm_model(config: DictConfig, model_registry: Registry, load_from: str
     return model
 
 
-def get_classification_layers(config: DictConfig, model_registry: Registry, prefix: str = "") -> list[str]:
-    """Return classification layer names by comparing two different number of classes models."""
+def get_classification_layers(
+    config: DictConfig,
+    model_registry: Registry,
+    prefix: str = "",
+) -> dict[str, dict[str, int]]:
+    """Return classification layer names by comparing two different number of classes models.
+
+    Args:
+        config (DictConfig): Config for building model.
+        model_registry (Registry): Registry for building model.
+        prefix (str): Prefix of model param name.
+            Normally it is "model." since OTXModel set it's nn.Module model as self.model
+
+    Return:
+        dict[str, dict[str, int]]
+        A dictionary contain classification layer's name and information.
+        Stride means dimension of each classes, normally stride is 1, but sometimes it can be 4
+        if the layer is related bbox regression for object detection.
+        Extra classes is default class except class from data.
+        Normally it is related with background classes.
+    """
     sample_config = deepcopy(config)
     modify_num_classes(sample_config, 5)
     sample_model_dict = build_mm_model(sample_config, model_registry, None).state_dict()
 
     modify_num_classes(sample_config, 6)
     incremental_model_dict = build_mm_model(sample_config, model_registry, None).state_dict()
 
-    return [
-        prefix + key for key in sample_model_dict if sample_model_dict[key].shape != incremental_model_dict[key].shape
-    ]
+    classification_layers = {}
+    for key in sample_model_dict:
+        if sample_model_dict[key].shape != incremental_model_dict[key].shape:
+            sample_model_dim = sample_model_dict[key].shape[0]
+            incremental_model_dim = incremental_model_dict[key].shape[0]
+            stride = incremental_model_dim - sample_model_dim
+            num_extra_classes = 6 * sample_model_dim - 5 * incremental_model_dim
+            classification_layers[prefix + key] = {"stride": stride, "num_extra_classes": num_extra_classes}
+    return classification_layers
```

Comment:
I guess I thought that I made a comment about this function in the last PR but it seems disappeared because of my mistake.

The gist of the comment I made at that time is that this function instantiates the deep learning model twice. You might feel this is nothing for CNNs (and almost our support models are extremally tiny fitted for edge devices). However, many deep learning models and especially modern ViTs require really huge memory usage. it is really reluctant for me to create a pair of models just to decide the classification layer.

After the design change about owning Python class for every OTX support model, we now has a room for the customization for each model at the code level. I recommend you to move to specifying the layer name. From a cost and utility perspective, it seems over engineering to me.

Reply:
What do you think of logic such as
If self.classification_layer_names is None:
    self.classification_layer = get_classification_layer()

I agree on your point about over-engineering and burden for system if model is too large, but 
auto detecting classification layer might helpful

Reply:
Let's look at it again later.

Reply:
Let's look at it again later.